### PR TITLE
Fix a11y nested interactive element in sub nav (Fixes #15005)

### LINKED ISSUE #15005: [a11y] sub nav component contains nested interactive elements
## Description

On mobile viewports, the sub navigation dropdown heading consists of two nested, interactive components (a link inside a button). This can cause issues for assisted technologies, and could also technically produce unexpected results in some browsers, since it's invalid HTML.

### Steps to reproduce

1. Open https://www.mozilla.org/en-US/firefox/new/
2. Resize to mobile viewport.
3. Inspect the navigation menu drop down

```html
<h2 class="c-sub-navigation-title is-summary">
  <button type="button" aria-controls="expand-csubnavigationcsubnavigationtitle-0" aria-expanded="false">
    <a href="/en-US/firefox/" data-link-type="nav" data-link-position="subnav" data-link-text="">
      <img class="c-sub-navigation-icon" src="https://www.mozilla.org/media/protocol/img/logos/firefox/logo.fedb52c912d6.svg" width="24" height="24" alt="">
        Firefox
    </a>
  </button>
</h2>
```

### Expected result

There should only be one interactive component.

### Actual result

image

### Environment

N/A

### PR DESCRIPTION
**Pull request by alexgibson**
## One-line summary

- Fixes sub navigation a11y issue where the drop down toggle on mobile viewports has a link inside a button.
- Updates all sub navigation instances to use the base macro.

## Issue / Bugzilla link

#15005

## Testing

- http://localhost:8000/en-US/firefox/new/
- http://localhost:8000/en-US/firefox/developer/
- http://localhost:8000/en-US/firefox/browsers/browser-history/
- http://localhost:8000/en-US/firefox/browsers/what-is-a-browser/
- http://localhost:8000/en-US/firefox/faq/
- http://localhost:8000/en-US/firefox/features/
- http://localhost:8000/en-US/products/vpn/

### CODE REVIEW

**Comment by janbrasna** [comment]
I think this also tangentially resolves #13065 — as now the macro properly [uses aria-current to denote](https://github.com/mozilla/bedrock/pull/14722#issuecomment-2183600689) the active page, so it clearly highlights when the user is looking at content for a given submenu item.

**Comment by alexgibson** [comment]
> I think this also tangentially resolves https://github.com/mozilla/bedrock/issues/13065

I believe that bug is more related to the sub navigation content, rather than the functionality?

**Comment by janbrasna** [comment]
I believe that was due to misunderstanding what is the role of the links, and an already fixed bug when the items in the submenu differed between each pages — leading the reporter into thinking these are CTAs defined for every page separately — while in reality it's a submenu that should stay consistent across the pages.

Now when the submenu items are all consistent, and with this PR also correctly marked up as a "current page" when viewed, it fixes the perception that the links are arbitrary CTAs rendering the OP request somewhat moot, when it's just a single common submenu.

**Comment by alexgibson** [comment]
There still seems to be some non-alignment in that bug about the expected content of the navigation items, so I'm not going to resolve it here. It could potentially be closed though, if those parties agree on what we have today being acceptable.

**Comment by janbrasna** [comment]
Right, that, asking if all these improvements make that a non-issue for the OP perhaps.

File: bedrock/products/templates/products/vpn/includes/subnav-refresh.html [hunk 1]
```diff
@@ -4,31 +4,29 @@
...
+    {
+      'text':  ftl('vpn-subnav-features'),
+      'href':  url('products.vpn.resource-center.landing'),
+      'cta_name': 'Learn about VPNs'
+    },
+    {
+      'text': ftl('sub-navigation-mobile'),
+      'href': url('products.vpn.features'),
+      'cta_name': 'Features'
+    },
```
Comment (janbrasna) [inline bedrock/products/templates/products/vpn/includes/subnav-refresh.html:25]:
```suggestion
  links=[
    {
      'text':  ftl('vpn-subnav-learn-about-vpns'),
      'href':  url('products.vpn.resource-center.landing'),
      'cta_name': 'Learn about VPNs'
    },
    {
      'text': ftl('vpn-subnav-features'),
      'href': url('products.vpn.features'),
      'cta_name': 'Features'
    },
```

**Review by reemhamz — APPROVED** [review]
Great job with this! Implemented pretty seamlessly. Thanks for this a11y fix :accessibility: r+

**Pull request merged**